src/validation_workflow/docker/validation_docker.py

```diff
@@ -29,23 +29,26 @@ def __init__(self, args: ValidationArgs) -> None:
         super().__init__(args)
 
     def download_artifacts(self) -> bool:
-
-        # Check if the docker daemon is running; or else, bring it up
-        if (self.is_container_daemon_running()):
+        try:
+            assert self.is_container_daemon_running(), 'Docker daemon is not running. Exiting the docker validation.'
 
             # STEP 1 . pull the images for OS and OSD
-            self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
-            self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
-
-            self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
-            self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
-            logging.info(f'the OS image ID is : {self.local_image_OS_id}')
-            logging.info(f'the OSD image ID is : {self.local_image_OSD_id} \n\n')
-
+            product_names = ["opensearch", "opensearch_dashboards"]
+            using_staging_artifact_only = 'staging' if self.args.using_staging_artifact_only else 'production'
+            get_image_id = lambda product: self.get_image_id(  # noqa: E731
+                self.get_artifact_image_name(product, using_staging_artifact_only),
+                self.args.version if not self.args.using_staging_artifact_only else ValidationArgs().stg_tag(product).replace(" ", ""))
+            self.image_ids = list(map(get_image_id, product_names))
+            logging.info(f'the opensearch image ID is : {self.image_ids[0]}')
+            logging.info(f'the opensearch-dashboards image ID is : {self.image_ids[1]} \n\n')
```

Comment:
This code should be

```python
self.images = DockerImages(["OpenSearch, "OpenSearch Dashboards"])
self.images.download()
```

You're missing a collection class `DockerImages` and an individual class `DockerImage`.